device/cuda/src/ambiguity_resolution/kernels/remove_tracks.cu

```diff
@@ -62,112 +65,127 @@ __global__ void remove_tracks(device::remove_tracks_payload payload) {
         payload.meas_to_remove_view);
     vecmem::device_vector<unsigned int> threads(payload.threads_view);
 
-    auto n_accepted_prev = (*payload.n_accepted);
+    const unsigned n_accepted_prev = *(payload.n_accepted);
+
+    __syncthreads();
+
     if (threadIndex == 0) {
         (*payload.n_accepted) -= *(payload.n_removable_tracks);
     }
 
     if (threadIndex < *(payload.n_meas_to_remove)) {
         sh_meas_ids[threadIndex] = meas_to_remove[threadIndex];
         sh_threads[threadIndex] = threads[threadIndex];
-    } else {
-        return;
+        is_valid_thread = true;
     }
 
-    const auto id = sh_meas_ids[threadIndex];
+    __syncthreads();
+
+    if (is_valid_thread) {
+
+        const auto id = sh_meas_ids[threadIndex];
+        is_duplicate = false;
 
-    bool is_duplicate = false;
-    for (unsigned int i = 0; i < threadIndex; ++i) {
-        if (sh_meas_ids[i] == id) {
-            is_duplicate = true;
-            break;
+        for (unsigned int i = 0; i < threadIndex; ++i) {
+            if (sh_meas_ids[i] == id) {
+                is_duplicate = true;
+                break;
+            }
         }
     }
-    if (is_duplicate) {
-        return;
-    }
 
-    const auto unique_meas_idx = meas_id_to_unique_id.at(id);
+    bool active = false;
 
-    // If there is only one track associated with measurement, the
-    // number of shared measurement can be reduced by one
-    const auto& tracks = tracks_per_measurement[unique_meas_idx];
-    auto track_status = track_status_per_measurement[unique_meas_idx];
+    if (!is_duplicate && is_valid_thread) {
 
-    auto trk_id = sorted_ids[n_accepted_prev - 1 - sh_threads[threadIndex]];
+        const auto id = sh_meas_ids[threadIndex];
+        const auto unique_meas_idx = meas_id_to_unique_id.at(id);
 
-    unsigned int worst_idx =
-        thrust::find(thrust::seq, tracks.begin(), tracks.end(), trk_id) -
-        tracks.begin();
+        // If there is only one track associated with measurement, the
+        // number of shared measurement can be reduced by one
+        const auto& tracks = tracks_per_measurement[unique_meas_idx];
+        auto track_status = track_status_per_measurement[unique_meas_idx];
 
-    track_status[worst_idx] = 0;
+        auto trk_id = sorted_ids[n_accepted_prev - 1 - sh_threads[threadIndex]];
```

Comment:
Even though CUDA debugging at the moment is on a fritz, I would still suggest writing:

```suggestion
        auto trk_id = sorted_ids.at(n_accepted_prev - 1 - sh_threads[threadIndex]);
```

So that issues could be found a bit more easily if/when things will go wrong. :thinking:

Reply:
Fixed.